helix-vcs/src/lib.rs

```diff
@@ -18,6 +19,7 @@ pub trait DiffProvider {
     /// The data is returned as raw byte without any decoding or encoding performed
     /// to ensure all file encodings are handled correctly.
     fn get_diff_base(&self, file: &Path) -> Option<Vec<u8>>;
+    fn get_current_head_name(&self, file: &Path) -> Option<Arc<ArcSwap<Arc<str>>>>;
```

Comment:
This should deref and just return `Option<&str>` by derefing or maybe `Option<String>`. The git backend already basically allocates a `String` then wraps it in this complicated Arc type.

Reply:
I suggested this type because when filewatching is implemented we can keep a copy of the `ArcSwap` in the filwatcher and swap it out when the repo changes on disk.

Reply:
That's a good point. Initially we also went for an `Option<String>`, but then we realized that once the filewatcher is in place, we will have to refactor the code and took the decision to implement it in a more future-proof way. Pascal's vision is for the diff provider itself to take care of the filewatching internally and therefore it's needed to return this complex type, to ensure that the consumers will be up to date with changes.